library/std/build.rs

```diff
@@ -183,3 +186,45 @@ fn main() {
         println!("cargo:rustc-cfg=reliable_f128_math");
     }
 }
+
+fn build_android() {
+    // Used to detect the value of the `__ANDROID_API__`
+    // builtin #define
+    const MARKER: &str = "BACKTRACE_RS_ANDROID_APIVERSION";
+    const ANDROID_API_C: &str = "
+BACKTRACE_RS_ANDROID_APIVERSION __ANDROID_API__
+";
+
+    // Create `android-api.c` on demand.
+    let out_dir = env::var_os("OUT_DIR").unwrap();
+    let android_api_c = Path::new(&out_dir).join("android-api.c");
+    std::fs::write(&android_api_c, ANDROID_API_C).unwrap();
+
+    let expansion = match cc::Build::new().file(&android_api_c).try_expand() {
+        Ok(result) => result,
+        Err(e) => {
+            eprintln!("warning: android version detection failed while running C compiler: {e}");
+            return;
+        }
+    };
+    let expansion = match std::str::from_utf8(&expansion) {
+        Ok(s) => s,
+        Err(_) => return,
+    };
```

Comment:
Suggestion: we could write these as let-else if we don't care about the error payload:

```rs
let Ok(expansion) = std::str::from_utf8(&expansion) else {
    return;
};
```

Similarly below.

Reply:
Good Suggestion! Done.